### PR TITLE
 fix(ivy): execute the optional begin and end methods of the rendererFactory

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: packages/core/test/render3/change_detection_spec.ts [hunk 1]
```diff
@@ -1043,4 +1044,40 @@ describe('change detection', () => {
 
   });
 
+  it('it should call begin and end when the renderer factory implements them', () => {
```
Comment (benlesh) [inline packages/core/test/render3/change_detection_spec.ts]: `it('it`  ... Two `it`s here.

File: packages/core/test/render3/change_detection_spec.ts [hunk 2]
```diff
@@ -1043,4 +1044,40 @@ describe('change detection', () => {
...
+    const log: string[] = [];
+
+    const testRendererFactory: RendererFactory3 = {
```
Comment (benlesh) [inline packages/core/test/render3/change_detection_spec.ts]: `testRendererFactory` doesn't appear to be used in this test. I'm not certain how `begin` and `end` are being called here.

**Review by benlesh — CHANGES_REQUESTED** [review]

**Comment by vicb** [comment]
http://test/OCL:207215700:BASE:207215924:1533270307413:4ba612c8

**Review by mhevery — APPROVED** [review]

File: packages/core/src/render3/view_ref.ts [hunk 3]
```diff
@@ -227,7 +228,16 @@ export class ViewRef<T> implements viewEngine_EmbeddedViewRef<T>, viewEngine_Int
    *
    * See {@link ChangeDetectorRef#detach detach} for more information.
```
Comment (trotyl) [inline packages/core/src/render3/view_ref.ts]: This link is wrong due to possibly copy&paste, also another one above. (Not caused by this PR)

**Review by benlesh — APPROVED** [review]

**Pull request closed without merging**